CorsixTH/Lua/HeatingController.lua

```diff
@@ -0,0 +1,241 @@
+--[[ Copyright (c) 2009 Peter "Corsix" Cawley
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of
+this software and associated documentation files (the "Software"), to deal in
+the Software without restriction, including without limitation the rights to
+use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+of the Software, and to permit persons to whom the Software is furnished to do
+so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE. --]]
+
+class "HeatingController"
+
+---@type HeatingController
+local HeatingController = _G["HeatingController"]
+
+local COLD_TOLERANCE_TEMP = 0.22
+local HOT_TOLERANCE_TEMP = 0.36
+local DEFAULT_DAYS_UNTIL_BOILER_BREAKDOWN = 200
+
+function HeatingController:HeatingController(world)
+  self.world = world
+  self:setRadiatorHeat(0.5)
+  self:resetBoilerBreakdownCounter()
+  self.acc_heating = 0
+  self.heating_broke = false
+  self.boiler_can_break = true
+  self.warmth_msg = false
+end
+
+function HeatingController:resetBoilerBreakdownCounter()
+  self.days_until_boiler_breakdown = self.world.map.level_config.gbv.DisasterLaunch
+  --if disasterLaunch not defined in Levelfile
+  if self.days_until_boiler_breakdown == nil then
+    self.days_until_boiler_breakdown = DEFAULT_DAYS_UNTIL_BOILER_BREAKDOWN
+  end
+end
+
+function HeatingController:checkHeatingFacilities(isPlayerHospital,day,averagePatiensWarmth, averageStaffWarmth)
+  if not self.warmth_msg and not self.heating_broke then
+    if day == 15 then
+      if averagePatiensWarmth < COLD_TOLERANCE_TEMP then
+        self:warningTooCold(isPlayerHospital)
+      elseif averagePatiensWarmth >= HOT_TOLERANCE_TEMP then
+        self:warningTooHot(isPlayerHospital)
+      end
+    end
+    -- Are the staff warm enough?
+    if day == 20 then
+      if averageStaffWarmth < COLD_TOLERANCE_TEMP then
+        if isPlayerHospital == true then
+          self.world.ui.adviser:say(_A.warnings.staff_very_cold)
+        end
+        self.warmth_msg = true
+      elseif averageStaffWarmth >= HOT_TOLERANCE_TEMP then
+        if isPlayerHospital == true then
+          self.world.ui.adviser:say(_A.warnings.staff_too_hot)
+        end
+        self.warmth_msg = true
+      end
+    end
+  end
+end
+
+function HeatingController:warningTooCold(isPlayerHospital)
```

Comment:
Remove the isPlayerHospital argument here; it's not useful. Just don't call this method if you don't want to display anything.

Actually I don't think this advisor stuff (or anything that depends on whether it's the player hospital) belongs in this class. I think the class should just expose whether it's too hot or cold.